virttest/utils_net.py

```diff
@@ -3510,7 +3510,8 @@ def get_host_ip_address(params=None, ip_ver="ipv4", linklocal=False):
         net_dev = get_default_gateway(iface_name=True)
     LOG.warning("No IP address of host was provided, using IP address"
                 " on %s interface", net_dev)
-    return get_ip_address_by_interface(net_dev, ip_ver, linklocal)
+    single_dev = net_dev.split('\n')[0].strip()
```

Comment:
Can we ensure in any case that net_dev.split('\n')[0] is our anticipated dev, rather than net_dev.split('\n')[1]?

Reply:
Normally, there is only 1 default gateway for one system. But there is time when the system has more than 1 interface connected and they are in the different subnet, which may cause "2 default gateways in one system". Just like below:
# ip route | grep default
default via 192.168.122.1 dev br0 
default via 192.168.0.1 dev enp4s0 proto dhcp metric 101 
In my case, I want to get the host ip address, actually, any interface is okay. So I choose the  net_dev.split('\n')[0], it can ensure getting a value no matter there is 1 or 2 default gateway.